_posts/pccisme/2020-08-31-pccisme-First-Reflection.md

```diff
@@ -0,0 +1,36 @@
+
+### Readings
+
+After I read the article - [Why Python?](https://inf380p.github.io/reading/why-python.html) by Dr. Hauser, I want to share some reflections and experience based on my learning journey.
+
+Before I started to learn Python, I learned the basic C++ from a course in undergraduate. When I was learning C++, I cannot feel that I want to use it to create some cool stuff by myself. For me, there were too many difficult knowledge points you need to understand before you can start building a simple program. After the semester, I have never opened "Dev-C++ (a highly recommended IDE for learning C++)" again. I would say that C++ avoided me from being a developer at the time.
+
+In 2013, I started learning Python by following some free online courses, but I just want to have an idea about why Python is so hot. It was like everyone is talking about how they can achieve with Python in ML and data science areas. Until 2019, I took the well-known course - [Programming for Everybody](https://www.coursera.org/learn/python) and finished all the exercises. I felt that Python is much easier to learn and implement my idea compared to C++, PHP, and Java. Later, I was confident in Python and was ready to try building something. Even though I don't create everything with Python, I can learn new technology faster because of my fundamental knowledge of Python. Therefore, my experience should support the statement that Python is a great language for beginners.
+
+Nevertheless, Python is not perfect and is not perfect for beginners, too. When I was trying to build a mobile app, I found that I am not familiar with the idea of object-oriented design. Also, I did not have much idea about what "Class," "Object," and "Method" mean. In contrast, I took a Java course this summer, and it explained these concepts in the very beginning. Furthermore, before I started learning JavaScript, I read some articles, and most of them discuss JavaScript is similar to Python. Yes, both of them are dynamic type language. Still, I was confused by different methods which I could not understand where they come from, especially when I was trying to interact with the Document Object Model. Despite frustrations, these made me noticed that I'd missed some essential concepts.
+
+Last, I still recommend Python to my friends who want to know more about programming. Once you feel that you can do something more with programming, it's the moment when your new journey begins.
+
+### Exercise
+
+I was studying the [conversion rules](https://dmitripavlutin.com/javascriptss-addition-operator-demystified/) in JavaScript, and I was surprised because I always convert integers to strings before printing in Python. 
+
+So, when I was doing the exercise, I actually tried to make sure that Python doesn't apply the same conversion rules in JavaScript and it would pop out the error messages if you try to do something as follows:
+
+```python
+print("Hello world" + 1)
```

Comment:
Great reflection. I thought I'd drop in here that Python can print non-string expressions, you just can't concatenate them with `+`. `print(1)` will know that the string representation of the integer 1 is "1". If you need to print this alongside strings, pass them as separate arguments to the print function: `print("Hello World", 1)`

On trinket, you may need to add the hashbang `#1/bin/python3` to the top of your trinket to get this behavior, since otherwise it might think you're trying to print a tuple.

We'll get to most of this eventually, but I thought you might enjoy learning some of it now.

Reply:
Hello Dr. Hauser,

Thank you for the feedback. Also, is there anything I need to update before merging?

Reply:
No. I'm merging first reflections today, so nothing else needed.